lib/bolt/inventory/group2.rb

```diff
@@ -190,25 +200,28 @@ def add_target(target)
 
         target['config'] = config_only_plugin(target['config'])
 
-        unless target.include?('alias')
-          return
-        end
-
-        aliases = target['alias']
-        aliases = [aliases] if aliases.is_a?(String)
-        unless aliases.is_a?(Array)
-          msg = "Alias entry on #{target['name']} must be a String or Array, not #{aliases.class}"
-          raise ValidationError.new(msg, @name)
-        end
+        if target.include?('alias')
+          aliases = target['alias']
+          aliases = [aliases] if aliases.is_a?(String)
+          unless aliases.is_a?(Array)
+            msg = "Alias entry on #{target['name']} must be a String or Array, not #{aliases.class}"
+            raise ValidationError.new(msg, @name)
+          end
 
-        aliases.each do |alia|
-          raise ValidationError.new("Invalid alias #{alia}", @name) unless alia =~ NAME_REGEX
+          aliases.each do |alia|
+            raise ValidationError.new("Invalid alias #{alia}", @name) unless alia =~ NAME_REGEX
 
-          if (found = @aliases[alia])
-            raise ValidationError.new(alias_conflict(alia, found, target['name']), @name)
+            if (found = @aliases[alia])
+              raise ValidationError.new(alias_conflict(alia, found, target['name']), @name)
+            end
+            @aliases[alia] = target['name']
           end
-          @aliases[alia] = target['name']
         end
+
+        raise ValidationError.new("Node #{target} does not have a name", @name) unless target['name']
```

Comment:
This seems like it will already be raised on line 173?  Are `if target['name'].nil?` and `unless target['name']` different?

Reply:
yeah, that seems safe to remove.